### PR TITLE
net-im/profanity: add use flag to add support of python plugins

### PR DESCRIPTION
**Pull request by jubalh**

### CODE REVIEW

File: net-im/profanity/profanity-0.14.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	test? ( dev-util/cmocka )
+    python? (
+	  $(python_gen_cond_dep '
+        dev-python/cython[${PYTHON_USEDEP}]
+      ')
+   )
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.14.0-ox-carbons.patch"
```
Comment (Flowdalic) [inline net-im/profanity/profanity-0.14.0-r1.ebuild:59]: Can't we use the GitHub patch URL, e.g.,https://github.com/profanity-im/profanity/commit/122434a276e21017ba294bda25164e27fd7cd731.patch,  to pull these patches via DISTDIR? This avoids having to place them in FILESDIR and then forgetting to remove them once the ebuild is dropped.

**Review by Flowdalic — CHANGES_REQUESTED** [review]

File: net-im/profanity/profanity-0.14.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{9..11} )
```
Comment (Flowdalic) [inline net-im/profanity/profanity-0.14.0-r1.ebuild:6]: I think Python 3.9 can be dropped now that 3.10 is the default

**Comment by jubalh** [comment]
@Flowdalic thanks for the review. I applied your suggestions! I don't think I would have forgotten to remove the patches but the method you described was actually new to me so I used it :)

File: net-im/profanity/profanity-0.14.0-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+PYTHON_COMPAT=( python3_{10..11} )
+inherit python-single-r1
+
+DESCRIPTION="A console based XMPP client inspired by Irssi"
+HOMEPAGE="https://profanity-im.github.io"
+SRC_URI="
+    https://github.com/profanity-im/profanity/releases/download/${PV}/${P}.tar.gz
+    https://github.com/profanity-im/profanity/commit/122434a.patch
+        -> ${PN}-ox-carbons.patch
```
Comment (Flowdalic) [inline net-im/profanity/profanity-0.14.0-r1.ebuild:14]: Please use $PN-0.14.0-foo-bar.patch as naming scheme for the patches.

Comment (jubalh) [reply net-im/profanity/profanity-0.14.0-r1.ebuild]: Done

**Comment by Flowdalic** [comment]
LGTM, but you may should squash the commits into one. I am currently abroad and can not test and merge this. Please ask my fellow proxy maintainers in IRC to test and merge this.

**Comment by jubalh** [comment]
> but you may should squash the commits into one

Really? But it's two different changes. Usually like to keep them apart.

> I am currently abroad and can not test and merge this. Please ask my fellow proxy maintainers in IRC to test and merge this.

Ok, thank you :)

**Comment by Flowdalic** [comment]
> Really? But it's two different changes. Usually like to keep them apart.

No strong opinion. I sometimes squash changes that would otherwise require "individual" revision bumps.

And strictly speaking, adding the python use flag would not require a revision bump and could be done on the existing ebuild. Which had the advantage that the diff would be immediately obvious.

But again, no strong opinion.

**Comment by jubalh** [comment]
> And strictly speaking, adding the python use flag would not require a revision bump and could be done on the existing ebuild

I see! I asked that on some IRC channel and got a "probably it needs one". That's why I did it during this commit :)

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**